manager/apis/app/v1alpha1/m4dapplication_webhook.go

```diff
@@ -68,6 +68,7 @@ func (r *M4DApplication) validateM4DApplicationSpec() []*field.Error {
 	var allErrs []*field.Error
 	specField := field.NewPath("spec").Child("data")
 	for i, dataSet := range r.Spec.Data {
+		dataSet := dataSet
```

Comment:
No one would understand it. Maybe use a different variable name and add a comment